packages/plumier/test/integration/application/action-result.spec.ts

```diff
@@ -26,4 +26,26 @@ describe("Redirect Action Result", () => {
                 expect(resp.header.location).toBe("/animal/hello")
             })
     })
-})
\ No newline at end of file
+
+    it("Should be able to set array value on setHeader", async () => {
+        class AnimalController {
+            @route.get()
+            index() {
+                return response
+                .json({foo: "bar"})
+                .setHeader('set-cookie',["foo=bar;","message=hello;"])
+            }
+        }
+        const plumier = await new Plumier()
+            .set(new RestfulApiFacility())
+            .set({ controller: AnimalController })
+            .set({ mode: "production" })
+            .initialize()
+        await supertest(plumier.callback())
+            .get("/animal/index")
+            .expect(200)
+            .expect((resp:supertest.Response) => {
+                expect(resp.header['set-cookie'].length).toBe(2)
```

Comment:
As a workaround you can check the Node.js version then `expect` differently for each version.

```typescript
        const cookie = response.header["set-cookie"]
        if (process.versions.node.substr(0, 1) === "8") //check if current node.js is version 8
            expect(cookie).toMatchObject(["foo=bar;,message=hello;"])
        else
            expect(cookie).toMatchObject(["foo=bar;", "message=hello;"])
```

Reply:
okay, let me try